### PR TITLE
fix(core): Prevent shared user details being saved alongside execution data

### PR DESCRIPTION
**Pull request by freyamade**

### CODE REVIEW

File: packages/cli/src/WorkflowExecuteAdditionalData.ts [hunk 1]
```diff
@@ -583,13 +584,19 @@ function hookFunctionsSave(parentProcessMode?: string): IWorkflowExecuteHooks {
 						}
 					}
 
+					// Although it is treated as IWorkflowBase here, it's being instantiated elsewhere with properties that may be sensitive
+					// As a result, we should remove any sensitive information before saving to the database
+					const { shared, ...workflowData } = this.workflowData as WorkflowEntity;
+
+					console.log(this.workflowData, '\n', workflowData);
```
Comment (netroy) [inline packages/cli/src/WorkflowExecuteAdditionalData.ts:591]: please remove the console.log

Comment (freyamade) [reply packages/cli/src/WorkflowExecuteAdditionalData.ts]: good catch ty

File: packages/cli/src/WorkflowExecuteAdditionalData.ts [hunk 2]
```diff
@@ -583,13 +584,28 @@ function hookFunctionsSave(parentProcessMode?: string): IWorkflowExecuteHooks {
 						}
 					}
 
+					// Although it is treated as IWorkflowBase here, it's being instantiated elsewhere with properties that may be sensitive
+					// As a result, we should create an IWorkflowBase object with only the data we want to save in it.
+					const pristineWorkflowData: IWorkflowBase = {
```
Comment (netroy) [inline packages/cli/src/WorkflowExecuteAdditionalData.ts:589]: you could use `lodash.pick` or `lodash.omit` to make this less verbose.

**Review by krynble — APPROVED** [review]
LGTM

**Comment by janober** [comment]
Got released with `n8n@0.214.0`

**Pull request merged**